website/content/en/docs/upgrading/v1-migration.md

```diff
@@ -176,233 +289,297 @@ kubectl patch customresourcedefinitions ec2nodeclasses.karpenter.k8s.aws -p "{\"
+     --set webhook.port=8443
+   ```
 
-* **Kubelet Configuration**: If you have multiple NodePools pointing to the same EC2NodeClass that have different kubeletConfigurations,
-then you have to manually add more EC2NodeClasses and point their NodePools to them. This will induce drift and you will have to roll your cluster.
-If you have multiple NodePools pointing to the same EC2NodeClass, but they have the same configuration, then you can proceed with the migration
-without having drift or having any additional NodePools or EC2NodeClasses configured.
+5. Rollback the Karpenter Controller:
 
-* **Remove kubelet annotation from NodePools**: During the upgrade process Karpenter will rely on the `compatibility.karpenter.sh/v1beta1-kubelet-conversion` annotation to determine whether to use the v1beta1 NodePool kubelet configuration or the v1 EC2NodeClass kubelet configuration. The   `compatibility.karpenter.sh/v1beta1-kubelet-conversion` NodePool annotation takes precedence over the EC2NodeClass Kubelet configuration when launching nodes. Remove the kubelet-configuration annotation (`compatibility.karpenter.sh/v1beta1-kubelet-conversion`) from your NodePools once you have migrated kubelet from the NodePool to the EC2NodeClass.
+   ```bash
+   # Service account annotation can be dropped when using pod identity
+   helm upgrade --install karpenter oci://public.ecr.aws/karpenter/karpenter --version ${KARPENTER_VERSION} --namespace "${KARPENTER_NAMESPACE}" --create-namespace \
+     --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"=${KARPENTER_IAM_ROLE_ARN} \
+     --set settings.clusterName=${CLUSTER_NAME} \
+     --set settings.interruptionQueue=${CLUSTER_NAME} \
+     --set controller.resources.requests.cpu=1 \
+     --set controller.resources.requests.memory=1Gi \
+     --set controller.resources.limits.cpu=1 \
+     --set controller.resources.limits.memory=1Gi \
+     --set webhook.enabled=true \
+     --set webhook.port=8443 \
+     --wait
+   ```
 
-Keep in mind that rollback, without replacing the Karpenter nodes, will not be supported to an earlier version of Karpenter once the annotation is removed. This annotation is only used to support the kubelet configuration migration path, but will not be supported in v1.1.
+## Before Upgrading to `v1.1.0`
 
-### Downgrading
+You've successfully upgraded to `v1.0`, but more than likely your manifests are still `v1beta1`.
+You can continue to apply these `v1beta1` manifests on `v1.0`, but support will be dropped in `v1.1`.
+Before upgrading to `v1.1+`, you will need to migrate your manifests over to `v1`.
 
-Once the Karpenter CRDs are upgraded to v1, conversion webhooks are needed to help convert APIs that are stored in etcd from v1 to v1beta1. Also changes to the CRDs will need to at least include the latest version of the CRD in this case being v1. The patch versions of the v1beta1 Karpenter controller that include the conversion wehooks include:
+#### General Migration
```

Comment:
Do we need a note about checking CRD status stored versions? IIRC the update will fail if the migration controller hasn't finished its work.

Reply:
I'm on the fence on rather this should go in the v1 upgrade guide, or just include it in v1.1. There's nothing actionable to do until you're upgrading to v1.1. I did leave a TODO in the doc on this, but I was thinking it could be decoupled from this PR.